cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/LineageOperationsProcessor.java

```diff
@@ -197,6 +197,10 @@ private void addMergeOperation(Set<String> stageInputs, Map<String, Operation> p
 
     List<String> fieldsForJoin = new ArrayList<>();
     for (String field : fields) {
+      // for joiner, the field name is prefixed with stage name, needs to calculate this name to query for the origin
+      // from other stages, but in the inputFields map we should use the original name since the actual name can be
+      // same from different previous stages
+      String actualField = field;
```

Comment:
No, it is different, `actualField` is always the short name without the stage prefix, for non-joiner stages, this will be the same as field. For joiner stage, the field is in format <stagename>.<fieldname>, this actual field will get reassigned to <fieldname> in later stages. And it will be used to query for `processedOperations` since the field name there never contains the stage prefix. Will update the comment about this